application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java

```diff
@@ -90,6 +100,14 @@ private void handleHelpThreadCreated(ThreadChannel threadChannel) {
             .queueAfter(5, TimeUnit.SECONDS);
     }
 
+    private static User getMentionedAuthorByMessage(Message message) {
+        return message.getMentions().getUsers().get(0);
+    }
+
+    private boolean isPostedByBot() {
+        return jda.getSelfUser().isBot();
```

Comment:
that code is incorrect and always true. ur testing whether the self-user (i.e. the bot) is a bot - thats always true.

instead, u want to test whether the user who posted the message is the self user.

Reply:
@Zabuzard  Im not sure if i understand this part, "self-user" is supposed to be logged in account? we're checking if the message that started thread is posted by Bot, if not then we can rely on `threadChannel.getOwnerIdLong()`.

Reply:
@Zabuzard i tried to find a method in the Message object like `message.getSelfUser()` and it doesn't exist, there is only `isBot()`. You told me previously to use JDA's `getSelfUser()` for this situation. tbh I feel confused now 😵

Reply:
@ankitsmt211  self-user is the logged in user, yes. i.e. tj-bot. we are not trying to check if the message was posted by _a bot_ (there are multiple bots), but that it was posted by tj-bot (self user).

@firasrg i said `threadChannel.getJda().getSelfUser()` and then `message.getAuthor().equals(selfUser)` (or compare their IDs)